src/test/java/com/aitusoftware/babl/websocket/KeyDecoderTest.java

```diff
@@ -163,11 +164,12 @@ private void assertKeyDecoded(final String httpRequest, final boolean shouldBeDe
     {
         final ByteBuffer buffer = ByteBuffer.wrap(httpRequest.getBytes(StandardCharsets.UTF_8));
         assertThat(keyDecoder.decode(
-            buffer, this::captureCharKey))
+            buffer, this::captureCharKey, this::captureRequestUri))
             .isEqualTo(shouldBeDecoded);
         if (shouldBeDecoded)
         {
             assertThat(capturedKey).isEqualTo(WEB_SOCKET_KEY.getBytes(StandardCharsets.UTF_8));
+            assertThat(capturedUri).isEqualTo("/".getBytes(StandardCharsets.UTF_8));
```

Comment:
Please add a test that demonstrates that the URI is only copied up to the first space character (as implemented in `KeyDecoder`).